samza-core/src/main/java/org/apache/samza/runtime/LocalApplicationRunner.java

```diff
@@ -441,15 +441,24 @@ public void afterStop() {
 
     @Override
     public void afterFailure(Throwable t) {
-      processors.removeIf(pair -> pair.getLeft().equals(processor));
-
+      // we need to close associated coordinator metadata store, although the processor failed
+      processors.forEach(sp -> {
```

Comment:
Won't this cause the current processor to not be stopped?

Do we also need to stop the metadata store in afterStop()?

Reply:
I think metadata store are closed before calling afterStop().

Reply:
@MabelYC It's not obvious to me how we can guarantee that all metadata stores are always closed. Can you document why you think metadata stores are always closed before calling afterStop?

Reply:
I double checked that the close loop contains some LI specific hookup with offspring. I updated the commit to stop the metadata store in afterStop() to make sure it got closed for sure.